// @phibase/common-v2/build/events/subjects.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Subjects = void 0;
var Subjects;
(function (Subjects) {
    // Common Domain
    // Auth Events
    Subjects["CustomerCreated"] = "user-customer:created";
    Subjects["CustomerUpdated"] = "user-customer:updated";
    Subjects["CustomerDeleted"] = "user-customer:deleted";
    Subjects["SalesPersonCreated"] = "user-salesperson:created";
    Subjects["SalesPersonUpdated"] = "user-salesperson:updated";
    Subjects["SalesPersonDeleted"] = "user-salesperson:deleted";
    Subjects["UserCreated"] = "user:created";
    Subjects["UserUpdated"] = "user:updated";
    Subjects["UserDeleted"] = "user:deleted";
    Subjects["UserLogin"] = "user:login";
    // Log Service
    Subjects["LogCreated"] = "log:created";
    // Item Service
    Subjects["ItemCreated"] = "item:created";
    Subjects["ItemUpdated"] = "item:updated";
    Subjects["ItemDeleted"] = "item:deleted";
    // Manufacture Domain
    // Plant Service
    Subjects["HardwareCreated"] = "hardware:created";
    Subjects["HardwareUpdated"] = "hardware:updated";
    Subjects["HardwareDeleted"] = "hardware:deleted";
    Subjects["StationCreated"] = "station:created";
    Subjects["StationUpdated"] = "station:updated";
    Subjects["StationDeleted"] = "station:deleted";
    Subjects["InstructionCreated"] = "instruction:created";
    Subjects["InstructionUpdated"] = "instruction:updated";
    Subjects["InstructionDeleted"] = "instruction:deleted";
    //Hardware Monitor Service
    Subjects["AndonCreated"] = "andon:created";
    Subjects["AndonUpdated"] = "andon:updated";
    Subjects["AndonDeleted"] = "andon:deleted";
    Subjects["AndonActivated"] = "andon:activated";
    Subjects["AndonDeactivated"] = "andon:deactivated";
    Subjects["MaintanceCreated"] = "maintance:created";
    Subjects["MaintanceUpdated"] = "maintance:updated";
    Subjects["MaintanceDeleted"] = "maintance:deleted";
    //Production Management Service
    Subjects["RecipeCreated"] = "recipe:created";
    Subjects["RecipeUpdated"] = "recipe:updated";
    Subjects["RecipeDeleted"] = "recipe:deleted";
    Subjects["WorkOrderCreated"] = "work-order:created";
    Subjects["WorkOrderUpdated"] = "work-order:updated";
    Subjects["WorkOrderDeleted"] = "work-order:deleted";
    Subjects["WorkOrderScanned"] = "work-order:scanned";
    Subjects["WorkOrderApproved"] = "work-order:approved";
    Subjects["InstructionFinished"] = "recipe-instruction:finished";
    Subjects["JobOrderApproved"] = "manufacture-job-order:approved";
    Subjects["ItemManufactureStarted"] = "item-manufacture:started";
    Subjects["ItemManufactureFinished"] = "item-manufacture:finished";
    Subjects["ItemManufactureRecycled"] = "item-manufacture:recycled";
    Subjects["ReworkTriggered"] = "rework:triggered";
    //Quality Control Service
    Subjects["QualityControlCreated"] = "quality-control:created";
    Subjects["QualityControlUpdated"] = "quality-control:updated";
    Subjects["QualityControlDeleted"] = "quality-control:deleted";
    Subjects["TestParameterCreated"] = "test-parameter:created";
    Subjects["TestParameterUpdated"] = "test-parameter:updated";
    Subjects["TestParameterDeleted"] = "test-parameter:deleted";
    // Distribution Domain
    // Stock
    Subjects["ItemStockChange"] = "stock:changed";
    //Delivery-Commerce
    Subjects["DelivToCommerceCreated"] = "delivery-commerce:created";
    Subjects["DelivToCommerceUpdated"] = "delivery-commerce:updated";
    // Warehouse Service
    Subjects["LocationCreated"] = "location:created";
    Subjects["LocationUpdated"] = "location:updated";
    Subjects["LocationDeleted"] = "location:deleted";
    Subjects["WarehouseCreated"] = "warehouse:created";
    Subjects["WarehouseUpdated"] = "warehouse:updated";
    Subjects["WarehouseDeleted"] = "warehouse:deleted";
    Subjects["MovingCreated"] = "moving:created";
    Subjects["MovingUpdated"] = "moving:updated";
    Subjects["MovingDeleted"] = "moving:deleted";
    //Inventory Service
    Subjects["InventoryCreated"] = "inventory:created";
    Subjects["InventoryUpdated"] = "inventory:updated";
    Subjects["InventoryDeleted"] = "inventory:deleted";
    //Picking Service
    Subjects["PickingCreated"] = "picking:created";
    Subjects["PickingUpdated"] = "picking:updated";
    Subjects["PickingDeleted"] = "picking:deleted";
    //Delivery Service
    Subjects["VehicleCreated"] = "vehicle:created";
    Subjects["VehicleUpdated"] = "vehicle:updated";
    Subjects["VehicleDeleted"] = "vehicle:deleted";
    Subjects["DeliveryCreated"] = "delivery:created";
    Subjects["DeliveryUpdated"] = "delivery:updated";
    Subjects["DeliveryDeleted"] = "delivery:deleted";
    // Commerce Domain
    // Request Service
    Subjects["RequestCreated"] = "request:created";
    Subjects["RequestUpdated"] = "request:updated";
    Subjects["RequestDeleted"] = "request:deleted";
    // Order Service
    Subjects["OrderCreated"] = "order:created";
    Subjects["OrderUpdated"] = "order:updated";
    Subjects["OrderDistributionUpdated"] = "order-dist:updated";
    Subjects["OrderManufactureUpdated"] = "order-manu:updated";
    Subjects["OrderDeleted"] = "order:deleted";
    // Payment Service
    Subjects["InvoiceCreated"] = "invoice:created";
    Subjects["InvoiceUpdated"] = "invoice:updated";
    Subjects["InvoiceDeleted"] = "invoice:deleted";
    Subjects["PaymentCreated"] = "payment:created";
    Subjects["PaymentUpdated"] = "payment:updated";
    Subjects["PaymentDeleted"] = "payment:deleted";
    //POS Service
    Subjects["TransactionCreated"] = "transaction:created";
    Subjects["TransactionUpdated"] = "transaction:updated";
    Subjects["TransactionDeleted"] = "transaction:deleted";
    //Receive Service
    Subjects["ReceiveCreated"] = "receive:created";
    Subjects["ReceiveUpdated"] = "receive:updated";
    Subjects["ReceiveDeleted"] = "receive:deleted";
    //Warranty Service
    Subjects["WarrantyCreated"] = "warranty:created";
    Subjects["WarrantyUpdated"] = "warranty:updated";
    Subjects["WarrantyDeleted"] = "warranty:deleted";
    // --------------------
    Subjects["NewDocument"] = "document:created";
    Subjects["DeleteDocument"] = "document:deleted";
})(Subjects = exports.Subjects || (exports.Subjects = {}));
